### PR TITLE
15314: use correct filename logic for new deliveries API

### PR DESCRIPTION
**Pull request by mkalish**
This PR updates the new delivery API to set the filename to be what the receiver would have gotten

Test Steps:
1. Run the deliveries API
```sh
curl --location 'http://localhost:7071/api/v1/waters/org/ignore/deliveries?receivingOrgSvcStatus=ACTIVE%2CTESTING' \
--header 'authentication-type: okta' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer dummy' \
--data '{
    "sort": {
        "direction": "DESC",
        "property": "createdAt"
    },
    "pagination": {
        "page": 1,
        "limit": 10
    },
    "filters": []
}'
```
2. Grab one the ids
3. Run the download report API
```sh
curl --location 'http://localhost:7071/api/history/report/{ID}' \
--header 'organization: ignore' \
--header 'Authorization: Bearer dummy'
```
4. Confirm the filename is the same

## Changes
- After fetching the rows from the db, processes the DeliveryHistory models to correctly set the filename

### Testing
- [x] Tested locally?
- [x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [x] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- https://github.com/CDCgov/prime-reportstream/issues/15314

### CODE REVIEW

**Comment by JFisk42** [comment]
I'm not sure if I misunderstood the test steps but I wasn't able to verify that the filenames are the same:

```
$ curl --location 'http://localhost:7071/api/v1/waters/org/ignore/deliveries?receivingOrgSvcStatus=ACTIVE%2CTESTING' \  
--header 'authentication-type: okta' \
--header 'Content-Type: application/json' \
--header 'Authorization: Bearer dummy' \
--data '{
    "sort": {
        "direction": "DESC",
        "property": "createdAt"
    },
    "pagination": {
        "page": 1,
        "limit": 10
    },
    "filters": []
}'
```
```
{
  "data" : [ {
  { <CLIPPED_ENTRIES> },
  {  
    "deliveryId" : "29799",
    "createdAt" : "2024-07-26T16:40:00.558Z",
    "expiresAt" : "2024-08-25T16:40:00.558Z",
    "receivingOrgSvcStatus" : "active",
    "reportId" : "78959e6f-1028-49c3-b892-acbabadcce8d",
    "topic" : "full-elr",
    "reportItemCount" : 2,
    "fileType" : "HL7_BATCH",
    "receiver" : "ignore.FULL_ELR",
    "fileName" : "ignore.FULL_ELR%2F78959e6f-1028-49c3-b892-acbabadcce8d.hl7"
  } ],
  "meta" : {
    "type" : "delivery_history",
    "totalCount" : 4198,
    "totalFilteredCount" : 4198,
    "totalPages" : 420,
    "nextPage" : 2
  }
}
```
```
$ curl --location 'http://localhost:7071/api/history/report/78959e6f-1028-49c3-b892-acbabadcce8d' \ 
--header 'organization: ignore' \     
--header 'Authorization: Bearer dummy'  
```
```
{
  "sent": 1722012000000,
  "via": "HL7_BATCH",
  "total": 2,
  "fileType": "HL7_BATCH",
  "type": "ELR",
  "reportId": "78959e6f-1028-49c3-b892-acbabadcce8d",
  "expires": 1724604000000,
  "sendingOrg": "",
  "receivingOrg": "ignore",
  "receivingOrgSvc": "FULL_ELR",
  "facilities": [],
  "actions": [],
  "displayName": "FULL_ELR",
  "content": <>,
  "fileName": "co.yml-78959e6f-1028-49c3-b892-acbabadcce8d-20240729170612.hl7",
  "mimeType": "application/hl7-v2"
}
```

File: prime-router/src/main/kotlin/history/db/DeliveryHistoryDatabaseAccess.kt
Comment (JFisk42) [inline prime-router/src/main/kotlin/history/db/DeliveryHistoryDatabaseAccess.kt:268]: I guess Kotlin was okay with it but having another results declared here feels weird with the variable two lines up and doing something like `results.results` a few lines below. Maybe historyResults or something like that?

**Review by JFisk42 — COMMENTED** [review]
Just a nitpick about a variable name and some confusion about the test steps.

**Review by JFisk42 — APPROVED** [review]
Approving after resolving my own slip on the tests steps.

**Review by penny-lischer — APPROVED** [review]
LGTM

File: prime-router/src/main/kotlin/history/db/DeliveryHistoryDatabaseAccess.kt
Comment (mkalish) [reply prime-router/src/main/kotlin/history/db/DeliveryHistoryDatabaseAccess.kt:268]: Good catch!  I wonder why intellij isn't giving me that shadowed varname warning 🤷🏻 .  Changed to `historyResults`

**Comment by mkalish** [comment]
@arnejduranovic This is ready for your review

File: prime-router/src/main/kotlin/history/db/DeliveryHistoryDatabaseAccess.kt
Comment (arnejduranovic) [inline prime-router/src/main/kotlin/history/db/DeliveryHistoryDatabaseAccess.kt:275]: Maybe a comment to describe what you are doing here since it may not be clear initially.

**Review by arnejduranovic — APPROVED** [review]
Looks good, one small comment. Please consider it before merging.

**Pull request merged**
